.dapper

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+params="$(getopt -o f:km:OC:sdqXS:hv -l file:,socket,mode:,no-out,build,directory:,shell,debug,quiet,keep,no-context,mount-suffix,target,help,version --name "$0" -- "$@")"
+eval set -- "$params"
+
+file=Dockerfile.dapper
+socket=false
+dockerargs=
+
+while true
+do
+    case "$1" in
+	--file|-f)
+	    file="$2"
+	    shift 2
+	    ;;
+	--socket|-k)
+	    socket=true
+	    shift
+	    ;;
+	--mode|-m)
+	    # Ignored, we only support bind
+	    shift 2
+	    ;;
+	--no-out|-O)
+	    # Ignored, we only support bind
+	    shift
+	    ;;
+	--build)
+	    # Not supported
+	    echo "$0 doesn't support --build yet"
+	    exit 1
+	    ;;
+	--directory|-C)
+	    cd "$2" || exit
+	    shift 2
+	    ;;
+	--shell|-s)
+	    command=bash
+	    dockerargs="${dockerargs} -it"
+	    DAPPER_ENV="${DAPPER_ENV} TERM"
+	    shift
+	    ;;
+	--debug|-d)
+	    shift
+	    set -x
+	    ;;
+	--quiet|-q|--keep|--no-context|-X|--help|-h)
+	    # Ignored
+	    shift
+	    ;;
+	--version|-v)
+	    echo Submariner Dapper
+	    exit 0
+	    ;;
+	--mount-suffix|-S)
+	    suffix=":$2"
+	    shift 2
+	    ;;
+	--)
+	    shift
+	    break
+	    ;;
+	*)
+	    echo "$0 doesn't support $1" >&2
+	    exit 1
+	    ;;
+    esac
+done
+
+[ -n "$command" ] && set -- "$command"
+
+gitid="$(git branch --show-current | tr / _)"
+gitid="${gitid:-$(git show --format=%h -s)}"
+container="$(basename "$(pwd)"):${gitid}"
+docker build -t "${container}" -f "${file}" --build-arg "BASE_BRANCH=${BASE_BRANCH:-devel}" .
+
+extract_var() {
+    docker inspect "$1" | sed -n -E "/\"$2=/{s/[[:space:]]+\"$2=([^\"]+)\",/\1/;p;q}"
```

Comment:
This should use `jq` but Podman doesn’t use exactly the same output format.

Reply:
Well, one less requirement :-)